src/Container.php

```diff
@@ -435,10 +449,10 @@ public function extend($resourceName, callable $callable)
         $resource = $this->getResource($key, true);
 
         $closure = function ($c) use ($callable, $resource) {
-            return $callable($resource->getInstance(), $c);
+            return $callable($resource->getInstance(true), $c);
         };
 
-        $this->set($key, $closure, $resource->isShared());
+        $this->set($key, $closure, $resource->isShared(), false, $resource->isLazy());
```

Comment:
@HLeithner, the `isLazy()` method is used to properly extend a service. Because the new service should be lazy too. You can see it here.

Also it could be useful for collect and showing information about services.

For example, Symfony has a command which shows full information about service (id, public, private, lazy and so on): https://symfony.com/doc/current/service_container/debug.html#detailed-info-about-a-single-service